bundles/org.openhab.binding.amplipi/src/main/java/org/openhab/binding/amplipi/internal/AmpliPiGroupHandler.java

```diff
@@ -0,0 +1,148 @@
+    }
+
+    @Override
+    public void initialize() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            bridgeHandler = (AmpliPiHandler) bridge.getHandler();
+            if (bridgeHandler != null) {
+                bridgeHandler.addStatusChangeListener(this);
+            } else {
+                throw new IllegalStateException("Bridge handler must not be null here!");
+            }
+            if (bridge.getStatus() == ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            }
+        } else {
+            throw new IllegalStateException("Bridge must not be null here!");
+        }
+    }
+
+    @Override
+    public void handleCommand(@NonNull ChannelUID channelUID, @NonNull Command command) {
+        if (command == RefreshType.REFRESH) {
+            // do nothing - we just wait for the next automatic refresh
+            return;
+        }
+        GroupUpdate update = new GroupUpdate();
+        switch (channelUID.getId()) {
+            case AmpliPiBindingConstants.CHANNEL_MUTE:
+                if (command instanceof OnOffType) {
+                    update.setMute(command == OnOffType.ON);
+                }
+                break;
+            case AmpliPiBindingConstants.CHANNEL_VOLUME:
+                if (command instanceof PercentType) {
+                    update.setVolDelta(AmpliPiUtils.percentTypeToVolume((PercentType) command));
+                }
+                break;
+            case AmpliPiBindingConstants.CHANNEL_SOURCE:
+                if (command instanceof DecimalType) {
+                    update.setSourceId(((DecimalType) command).intValue());
+                }
+                break;
+        }
+        if (bridgeHandler != null) {
+            String url = bridgeHandler.getUrl() + "/api/groups/" + id;
+            StringContentProvider contentProvider = new StringContentProvider(gson.toJson(update));
+            try {
+                ContentResponse response = httpClient.newRequest(url).method(HttpMethod.PATCH)
+                        .content(contentProvider, "application/json").send();
+                if (response.getStatus() != HttpStatus.OK_200) {
+                    logger.error("AmpliPi API returned HTTP status {}.", response.getStatus());
+                    logger.debug("Content: {}", response.getContentAsString());
+                } else {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (InterruptedException | TimeoutException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "AmpliPi request failed: " + e.getMessage());
+            } catch (ExecutionException e) {
+                logger.error("HTTP request to AmpliPi failed: {}.", e.getMessage());
```

Comment:
It doesn't seem resonable to log to error when the network fails.

Reply:
If there's a network issue, we run into the `TimeoutException` which is caught above.
I am not exactly sure in which situations an `ExecutionException` exactly occurs - that's why I rather considered it a bug situation as long as it isn't better handled. Do you happen to know when we can expect the `ExecutionException` from the httpClient?

Reply:
Interesting question. I always assumed that a `ExecutionException` can be caused by a network failure. I did a quick search and couldn't find any official documentation, but I found some stack traces showing this (in an OH context):
```
Caused by: java.util.concurrent.ExecutionException: java.net.ConnectException: Connection refused
at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)
...
```
I would handle this as a network failure.

Reply:
I am not sure. A "Connection refused" means that the network is available, but that the contacted device did not accept the request at all - meaning that the request was invalid (e.g. by using a wrong port), so it might be due to a bug in the code.
But it's ok for me to treat it like a network issue - users will complain, if their things go OFFLINE for an ununderstood reason anyhow. I've changed the code accordingly.

Reply:
Or the host is available, but the service is down. I think you can't clearly differentiate between temprorary or fatal errors here.